### PR TITLE
Read SkSL from json asset

### PR DESCRIPTION
**Pull request by liyuqian**
Fixes https://github.com/flutter/flutter/issues/55219

### CODE REVIEW

**Comment by liyuqian** [comment]
@jonahwilliams : let's forget about iOS Metal for now until developers ask for it. This should unblock us on iOS32 and Android. The Metal iOS would still load those SkSLs but they probably won't eliminate the compilation jank for now.

File: shell/common/persistent_cache.cc [hunk 1]
```diff
@@ -4,10 +4,13 @@
 
 #include "flutter/shell/common/persistent_cache.h"
 
+#include <rapidjson/document.h>
```
Comment (chinmaygarde) [inline shell/common/persistent_cache.cc:7]: Nit: Square brackets for system headers and quotes for others. This goes at the bottom with the others.

File: shell/common/persistent_cache.cc [hunk 2]
```diff
@@ -103,21 +106,34 @@ static std::shared_ptr<fml::UniqueFD> MakeCacheDirectory(
...
+  }
+  const std::string& data_string = decode_result.second;
+  return SkData::MakeWithCopy(data_string.data(), data_string.length());
+}
+
+sk_sp<SkData> ParseBase64(const std::string& input) {
+  SkBase64 decoder;
+  auto error = decoder.decode(input.c_str(), input.length());
+  if (error != SkBase64::Error::kNoError) {
+    FML_LOG(ERROR) << "Base64 can't decode: " << input;
```
Comment (chinmaygarde) [inline shell/common/persistent_cache.cc:123]: Nit: Also dump the error maybe?

File: shell/common/persistent_cache.h [hunk 3]
```diff
@@ -73,6 +73,7 @@ class PersistentCache : public GrContextOptions::PersistentCache {
   static void MarkStrategySet() { strategy_set_ = true; }
 
   static constexpr char kSkSLSubdirName[] = "sksl";
+  static constexpr char kAssetFileName[] = "shaders.json";
```
Comment (chinmaygarde) [inline shell/common/persistent_cache.h:76]: This seems like an all too common name and may collide with other assets. How about `io.flutter.shaders.json` or something more indicative of its purpose?

File: shell/common/persistent_cache.cc (hunk 1 above)
Comment (liyuqian) [reply shell/common/persistent_cache.cc]: Done.

File: shell/common/persistent_cache.cc (hunk 2 above)
Comment (liyuqian) [reply shell/common/persistent_cache.cc:124]: Done.

File: shell/common/persistent_cache.h (hunk 3 above)
Comment (liyuqian) [reply shell/common/persistent_cache.h]: Done.

**Pull request merged**